specification/compute/resource-manager/Microsoft.Compute/preview/2018-06-01/gallery.json

```diff
@@ -0,0 +1,1120 @@
+        }
+      ],
+      "description": "Specifies information about the gallery that you want to create or update."
+    },
+    "GalleryProperties": {
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "The description of this gallery resource."
+        },
+        "identifier": {
+          "$ref": "#/definitions/GalleryIdentifier"
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state, which only appears in the response."
+        }
+      },
+      "description": "Describes the properties of a gallery."
+    },
+    "GalleryIdentifier": {
+      "properties": {
+        "uniqueName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The unique name of the gallery"
+        }
+      }
+    },
+    "GalleryImage": {
+      "properties":{
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/GalleryImageProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Specifies information about the gallery image that you want to create or update."
+    },
+    "GalleryImageProperties": {
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "The description of this gallery image resource."
+        },
+        "eula": {
+          "type": "string",
+          "description": "The Eula agreement for the gallery image."
+        },
+        "privacyStatementUri": {
+          "type": "string",
+          "description": "The privacy statement uri."
+        },
+        "releaseNoteUri": {
+          "type": "string",
+          "description": "The release note uri."
+        },
+        "osType": {
+          "type": "string",
+          "description": "This property allows you to specify the type of the OS that is included in the disk if creating a VM from user-image or a specialized VHD. <br><br> Possible values are: <br><br> **Windows** <br><br> **Linux**",
+          "enum": [
+            "Windows",
+            "Linux"
+          ],
+          "x-ms-enum": {
+            "name": "OperatingSystemTypes",
+            "modelAsString": false
+          }
+        },
+        "osState": {
+          "type": "string",
+          "description": "The OS State.",
+          "enum": [
+            "Generalized",
+            "Specialized"
+          ],
+          "x-ms-enum": {
+            "name": "OperatingSystemStateTypes",
+            "modelAsString": false
+          }
+        },
+        "endOfLifeDate": {
+          "type": "string",
+          "format": "date",
+          "description": "The end of life of this gallery image."
+        },
+        "identifier": {
+          "$ref": "#/definitions/GalleryImageIdentifier"
+        },
+        "recommended": {
```

Comment:
can we make the property name more intuitive? may be hard now as its deployed, but still checking. recommendedMachineConfiguration may have been better.

Reply:
Yes, I agree with you. Renaming the property name may have been better. But I simply followed our API spec which the decision has been made between our team and the PM. Also, the code has been done and the service has been deployed. So it might be hard to make the change now..